src/map/pet.cpp

```diff
@@ -1646,44 +1637,42 @@ void read_petdb()
 				continue;
 			}
 
-			pet_db[j].class_ = nameid;
-			safestrncpy(pet_db[j].name,str[1],NAME_LENGTH);
-			safestrncpy(pet_db[j].jname,str[2],NAME_LENGTH);
-			pet_db[j].itemID=atoi(str[3]);
-			pet_db[j].EggID=atoi(str[4]);
-			pet_db[j].AcceID=atoi(str[5]);
-			pet_db[j].FoodID=atoi(str[6]);
-			pet_db[j].fullness=atoi(str[7]);
-			pet_db[j].hungry_delay=atoi(str[8])*1000;
-			pet_db[j].r_hungry=atoi(str[9]);
-			if( pet_db[j].r_hungry <= 0 )
-				pet_db[j].r_hungry=1;
-			pet_db[j].r_full=atoi(str[10]);
-			pet_db[j].intimate=atoi(str[11]);
-			pet_db[j].die=atoi(str[12]);
-			pet_db[j].capture=atoi(str[13]);
-			pet_db[j].speed=atoi(str[14]);
-			pet_db[j].s_perfor=(char)atoi(str[15]);
-			pet_db[j].talk_convert_class=atoi(str[16]);
-			pet_db[j].attack_rate=atoi(str[17]);
-			pet_db[j].defence_attack_rate=atoi(str[18]);
-			pet_db[j].change_target_rate=atoi(str[19]);
-			pet_db[j].pet_script = NULL;
-			pet_db[j].pet_loyal_script = NULL;
+			struct s_pet_db* pet = &pet_db_data[nameid];
+
+			pet->class_ = nameid;
+			safestrncpy(pet->name,str[1],NAME_LENGTH);
+			safestrncpy(pet->jname,str[2],NAME_LENGTH);
+			pet->itemID=atoi(str[3]);
+			pet->EggID=atoi(str[4]);
+			pet->AcceID=atoi(str[5]);
+			pet->FoodID=atoi(str[6]);
+			pet->fullness=atoi(str[7]);
+			pet->hungry_delay=atoi(str[8])*1000;
+			pet->r_hungry=atoi(str[9]);
+			if( pet->r_hungry <= 0 )
+				pet->r_hungry=1;
+			pet->r_full=atoi(str[10]);
+			pet->intimate=atoi(str[11]);
+			pet->die=atoi(str[12]);
+			pet->capture=atoi(str[13]);
+			pet->speed=atoi(str[14]);
+			pet->s_perfor=(char)atoi(str[15]);
+			pet->talk_convert_class=atoi(str[16]);
+			pet->attack_rate=atoi(str[17]);
+			pet->defence_attack_rate=atoi(str[18]);
+			pet->change_target_rate=atoi(str[19]);
+			pet->pet_script = NULL;
```

Comment:
can we use nullptr instead NULL ?, nullptr is safer in few case and there is no need to stick with NULL since it's not a C compiler anymore.

Reply:
yes of course since those are all .cpp files.